wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/FileUtil.java

```diff
@@ -74,21 +74,17 @@ public static boolean validateFileType(final FileItemWrap newFile, final List<St
 		// If extension match is unsucessful, then move to fileMimes list
 		boolean mimeMatch = false;
 		if (!extMatch && fileMimes.size() > 0) {
-			try {
-				final String mimeType = getFileMimeType(newFile);
-				LOG.debug("File mime-type is: " + mimeType);
-				for (String fileMime : fileMimes) {
-					if (StringUtils.equals(mimeType, fileMime)) {
+			final String mimeType = getFileMimeType(newFile);
+			LOG.debug("File mime-type is: " + mimeType);
+			for (String fileMime : fileMimes) {
+				if (StringUtils.equals(mimeType, fileMime)) {
+					mimeMatch = true;
```

Comment:
the function returns `true` if `mimeMatch` == `true` so you can significantly reduce cyclometric complexity by returning `true` rather than setting the member. This goes for all other of same type in this nested if group.